pkg/compactor/blocks_cleaner.go

```diff
@@ -56,17 +56,18 @@ type BlocksCleaner struct {
 	lastOwnedUsers []string
 
 	// Metrics.
-	runsStarted                 prometheus.Counter
-	runsCompleted               prometheus.Counter
-	runsFailed                  prometheus.Counter
-	runsLastSuccess             prometheus.Gauge
-	blocksCleanedTotal          prometheus.Counter
-	blocksFailedTotal           prometheus.Counter
-	blocksMarkedForDeletion     prometheus.Counter
-	tenantBlocks                *prometheus.GaugeVec
-	tenantMarkedBlocks          *prometheus.GaugeVec
-	tenantPartialBlocks         *prometheus.GaugeVec
-	tenantBucketIndexLastUpdate *prometheus.GaugeVec
+	runsStarted                    prometheus.Counter
+	runsCompleted                  prometheus.Counter
+	runsFailed                     prometheus.Counter
+	runsLastSuccess                prometheus.Gauge
+	blocksCleanedTotal             prometheus.Counter
+	blocksFailedTotal              prometheus.Counter
+	blocksMarkedForDeletion        prometheus.Counter
+	partialBlocksMarkedForDeletion prometheus.Counter
```

Comment:
After looking at `blocksMarkedForDeletion` closely I think we should reuse the `blocksMarkedForDeletionName` metric name and introduce new `reason="partial"` label. We can keep this counter, but initialize it similar to `blocksMarkedForDeletion` except with different reason.